src/backend/PostgresBackend.h

```diff
@@ -16,87 +16,118 @@ class PostgresBackend : public BackendInterface
     std::shared_ptr<PgPool> pgPool_;
     mutable PgQuery writeConnection_;
     mutable bool abortWrite_ = false;
-    mutable boost::asio::thread_pool pool_{16};
     uint32_t writeInterval_ = 1000000;
     uint32_t inProcessLedger = 0;
     std::unordered_set<std::string> successors_;
 
 public:
-    PostgresBackend(boost::json::object const& config);
+    PostgresBackend(
+        boost::asio::io_context& ioc,
+        boost::json::object const& config);
 
     std::optional<uint32_t>
-    fetchLatestLedgerSequence() const override;
+    fetchLatestLedgerSequence(boost::asio::yield_context& yield) const override;
 
     std::optional<ripple::LedgerInfo>
-    fetchLedgerBySequence(uint32_t sequence) const override;
+    fetchLedgerBySequence(uint32_t sequence, boost::asio::yield_context& yield)
+        const override;
 
     std::optional<ripple::LedgerInfo>
-    fetchLedgerByHash(ripple::uint256 const& hash) const override;
+    fetchLedgerByHash(
+        ripple::uint256 const& hash,
+        boost::asio::yield_context& yield) const override;
 
     std::optional<Blob>
-    doFetchLedgerObject(ripple::uint256 const& key, uint32_t sequence)
-        const override;
+    doFetchLedgerObject(
+        ripple::uint256 const& key,
+        uint32_t sequence,
```

Comment:
Consider changing to `std::uint32_t const` wherever possible. I won't highlight every instance.